conans/test/functional/old/scm_test.py

```diff
@@ -508,34 +508,45 @@ class MyLib(ConanFile):
         self.assertIn(commit, content)
 
     def test_delegated_python_code(self):
-        conanfile = """
-from conans import ConanFile
-from recipe_code import get_commit
-
-class MyLib(ConanFile):
-    name = "issue"
-    version = "3831"
-    scm = {'type': 'git', 'url': 'url', 'revision': get_commit()}
-"""
-
+        client = TestClient()
         code_file = """
 import os
-from conans.client.tools.scm import Git
+from conans.tools import Git
+from conans import ConanFile
 
 def get_commit():
-    here = os.path.dirname(__file__)
+    here = os.getcwd()
```

Comment:
It is not exactly the same... as a test (someone can look at it as an example) I prefer the previous behavior.

Reply:
With ``__file__`` it fails. The file of the python require is located in a different place than the repo